ui/components/app/signature-request/signature-request.js

```diff
@@ -0,0 +1,406 @@
+  const conversionRate = useNativeCurrencyAsPrimaryCurrency
+    ? null
+    : conversionRateSelected;
+  const subjectMetadata = useSelector(getSubjectMetadata);
+  // not forwarded to component
+  const allAccounts = useSelector(accountsWithSendEtherInfoSelector);
+  const { address, balance, name } =
+    getAccountByAddress(allAccounts, from) || {};
+
+  ///: BEGIN:ONLY_INCLUDE_IN(build-mmi)
+  // Used to show a warning if the signing account is not the selected account
+  // Largely relevant for contract wallet custodians
+  const selectedAccount = useSelector(getSelectedAccount);
+  ///: END:ONLY_INCLUDE_IN
+
+  let cancel;
+  let sign;
+
+  if (type === MESSAGE_TYPE.PERSONAL_SIGN) {
+    cancel = cancelPersonalMessage;
+    sign = signPersonalMessage;
+  } else if (type === MESSAGE_TYPE.ETH_SIGN_TYPED_DATA) {
+    cancel = cancelTypedMessage;
+    sign = signTypedMessage;
+  } else if (type === MESSAGE_TYPE.ETH_SIGN) {
+    cancel = cancelMessage;
+    sign = signMessage;
+  }
+
+  const memoizedParseMessage = memoize(() => {
+    const { message, domain = {}, primaryType, types } = JSON.parse(data);
+    const sanitizedMessage = sanitizeMessage(message, primaryType, types);
+    return { sanitizedMessage, domain, primaryType };
+  });
+
+  const getNetworkName = (providerType, nickName) => {
+    switch (providerType) {
+      case NETWORK_TYPES.MAINNET:
+        return t('mainnet');
+      case NETWORK_TYPES.GOERLI:
+        return t('goerli');
+      case NETWORK_TYPES.SEPOLIA:
+        return t('sepolia');
+      case NETWORK_TYPES.LINEA_TESTNET:
+        return t('lineatestnet');
+      case NETWORK_TYPES.LOCALHOST:
+        return t('localhost');
+      default:
+        return nickName || t('unknownNetwork');
+    }
+  };
+
+  const handleCancelAll = () => {
+    dispatch(
+      showModal({
+        name: 'REJECT_TRANSACTIONS',
+        onSubmit: async () => {
+          await dispatch(cancelMsgs(valuesFor(unconfirmedMessagesList)));
+          dispatch(clearConfirmTransaction());
+          history.push(mostRecentOverviewPage);
+        },
+        unapprovedTxCount: unapprovedMessagesCount,
+        isRequestType: true,
+      }),
+    );
+  };
+
+  const {
+    sanitizedMessage,
+    domain: { verifyingContract },
+    primaryType,
+  } = memoizedParseMessage();
+  const rejectNText = t('rejectRequestsN', [unapprovedMessagesCount]);
+  const currentNetwork = getNetworkName(providerName, providerNickName);
+
+  const balanceInBaseAsset = conversionRate
+    ? formatCurrency(
+        getValueFromWeiHex({
+          value: balance,
+          fromCurrency: nativeCurrency,
+          toCurrency: currentCurrency,
+          conversionRate,
+          numberOfDecimals: 6,
+          toDenomination: EtherDenomination.ETH,
+        }),
+        currentCurrency,
+      )
+    : new Numeric(balance, 16, EtherDenomination.WEI)
+        .toDenomination(EtherDenomination.ETH)
+        .round(6)
+        .toBase(10)
+        .toString();
```

Comment:
It will be nice to have all of above code before render and group methods above them.